app/merger.go

```diff
@@ -53,9 +53,10 @@ func (smartMerger) Merge(reports []report.Report) report.Report {
 		c <- r
 	}
 	for ; l > 1; l-- {
-		go func(left, right report.Report) {
+		left, right := <-c, <-c
+		go func() {
 			c <- left.Merge(right)
-		}(<-c, <-c)
+		}()
```

Comment:
This one's releated. See https://github.com/weaveworks/scope/pull/2026/commits/4007a902a264e5ff2c3be6b269ade515c9c1c145

Adding the new topologies put the `Report` struct over a critical size, such that passing two of them by value in a single function call causes a panic (something about running out of stack during goroutine startup).